lib/rubygems/sigstore/cert_extensions.rb

```diff
@@ -0,0 +1,25 @@
+module Gem::Sigstore::CertExtensions
+  def method_missing(method, *args, &block)
```

Comment:
This is syntactic sugar that I'm trying out.  We're exposing a certificate instance's extensions as attributes.  If someone calls `cert#some_unknown_extension`, then NoMethodError will be raised as usual.  We see an example of this in `CertChain#retrieve_issuer_cert`, above.

I'm on the fence about this, tbh.  We could very well do with the `#extension("extension name")` method bellow, sticking to a more explicit pattern.  If we do this, then there is no need to maintain our `@extension_hash`'s keys in snake case.